apple/swift-docc
# Include suggested solution messages in main diagnostic message when writing diagnostics to console

Bug/issue #, if applicable: https://github.com/apple/swift-docc/issues/478 <rdar://105985393>

## Summary

This works around a limitation that the messages of suggested solutions aren't written to console by including them in the main diagnostic message. It also updates the new link resolution suggestions to describe their replacement so that a reader can identify which described solution corresponds to which "fixit".

I will make more changes in this area but wanted to open this PR first so that we don't regress — by displaying less information to the developer — with the more structured diagnostics.

## Dependencies

None

## Testing

This can be tested with any diagnostic with solutions. This describes how to test it with a link resolution diagnostic.

In any documentation project with some symbols requiring disambiguation. 

1. Enable the hierarchy based link resolver
```
defaults write -g DocCUseHierarchyBasedLinkResolver -bool true
```
2. Find a link with disambiguation and remove the disambiguation.

3. Build documentation for the project and pass the `--emit-fixits` flag to DocC in the build.

The descriptions about which disambiguation to add to refer to which symbol should be included in the error message. 

Here's one example from TestBundle.docc

```
/path/to/swift-docc/Tests/SwiftDocCTests/Test Bundles/TestBundle.docc/documentation/myprotocol.md:11:41: warning: Topic reference 'MyUnresolvedSymbol' couldn't be resolved. Reference at '/MyKit/MyProtocol' can't resolve 'MyUnresolvedSymbol'. No similar pages. Replace 'MyUnresolvedSymbol' with 'Discussion'.
/path/to/swift-docc/Tests/SwiftDocCTests/Test Bundles/TestBundle.docc/documentation/myprotocol.md:11:41: note: Available children: Discussion.
/path/to/swift-docc/Tests/SwiftDocCTests/Test Bundles/TestBundle.docc/documentation/myprotocol.md:11:43-11:61: fixit: Discussion
```

Note the "Replace 'MyUnresolvedSymbol' with 'Discussion'." part of the warning message.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
